flask/helpers.py

```diff
@@ -268,40 +268,40 @@ def external_url_handler(error, endpoint, values):
     """
     appctx = _app_ctx_stack.top
     reqctx = _request_ctx_stack.top
+
     if appctx is None:
-        raise RuntimeError('Attempted to generate a URL without the '
-                           'application context being pushed. This has to be '
-                           'executed when application context is available.')
+        raise RuntimeError(
+            'Attempted to generate a URL without the application context being'
```

Comment:
I started doing it this way because I removed the space by mistake a lot more often when it appeared at the end.